src/snowflake/cli/_plugins/nativeapp/entities/application.py

```diff
@@ -612,11 +609,14 @@ def execute_post_deploy_hooks(self):
                     database_name=app_name,
                 )
 
-    @staticmethod
     @contextmanager
-    def use_application_warehouse(
-        app_warehouse: Optional[str],
-    ):
+    def use_application_warehouse(self):
+        model = self._entity_model
+        ctx = self._workspace_ctx
+        app_warehouse = (
+            model.meta and model.meta.warehouse and to_identifier(model.meta.warehouse)
+        ) or to_identifier(ctx.default_warehouse)
```

Comment:
I'm eventually gonna add properties to the entity model classes to make common operations simpler once these simplification PRs are done

(TBH this `to_identifier` should probably be done in `SqlExecutor.use_warehouse()` anyways)

Reply:
Soon this should be removed from here and moved to the facade

Reply:
yeah good point. I didn't want to make any changes like that in this current series of PRs, to make them simpler to review

Reply:
yeah agreed, just pointing out the identifier discussion is short lived, that's all (FWIW, the current invariant is the facade converts to a valid identifier).

Reply:
That being said, I do like Michel's suggestion. Any particular reason we wouldn't want to extract some kind of warehouse property here? It would be useful even post facade conversion.

Reply:
yeah i do want to do it, just not as part of this PR (since there are other places where this pattern is used, and I'd rather address them all once in a separate PR)